### PR TITLE
Oracle GoldenGate

### PR DESCRIPTION
**Pull request by sbalousek**
- Initial files to support containerized execution of Oracle GoldenGate 23

### CODE REVIEW

File: OracleGoldenGate/23/bin/deployment-init.py
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-init.py:5]: The spaces between the letters should be removed for accessibility purposes.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:2]: This should be `2024`

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:6]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:11]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:39]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:64]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:79]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:97]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:109]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:128]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:148]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:158]: Please remove the spaces.

File: OracleGoldenGate/23/bin/deployment-main.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/deployment-main.sh:159]: The correct way to format NGINX is in all capitals as per their trademark documentation: https://www.f5.com/company/policies/trademarks

File: OracleGoldenGate/23/bin/healthcheck
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/healthcheck:2]: This should be `2024`

File: OracleGoldenGate/23/bin/healthcheck
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/healthcheck:5]: Please remove the spaces

File: OracleGoldenGate/23/bin/replace-variables.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/replace-variables.sh:2]: This should be `2024`

File: OracleGoldenGate/23/bin/replace-variables.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/bin/replace-variables.sh:6]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:2]: This should be `2024`

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:12]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:17]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:26]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:38]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:49]: Please remove the spaces

File: OracleGoldenGate/23/install-deployment.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-deployment.sh:64]: Please remove the spaces

File: OracleGoldenGate/23/install-prerequisites.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-prerequisites.sh:2]: This should be `2024`

File: OracleGoldenGate/23/install-prerequisites.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-prerequisites.sh:6]: Please remove the spaces

File: OracleGoldenGate/23/install-prerequisites.sh
Comment (Djelibeybi) [inline OracleGoldenGate/23/install-prerequisites.sh:26]: I suggest changing this to `dnf -y module enable` (to make it faster) then add `nginx` and `python39` to the `packages` array so they get installed by the following command. You should also explicitly specify the NGINX module version as it could possibly change without notice in the future (unlikely, I admit, but possible nonetheless).

File: OracleGoldenGate/23/nginx/nginx.conf
Comment (Djelibeybi) [inline OracleGoldenGate/23/nginx/nginx.conf:1]: This should be `2024`

File: OracleGoldenGate/23/nginx/nginx.conf
Comment (Djelibeybi) [inline OracleGoldenGate/23/nginx/nginx.conf:5]: The correct way to format NGINX is in all capitals as per their trademark documentation (see other comment for the URL). While I'm being pedantic, "reverse proxy" shouldn't be capitalised here either. :)

File: OracleGoldenGate/23/nginx/ogg-headers.conf
Comment (Djelibeybi) [inline OracleGoldenGate/23/nginx/ogg-headers.conf:1]: This should be `2024`

File: OracleGoldenGate/23/nginx/ogg.conf
Comment (Djelibeybi) [inline OracleGoldenGate/23/nginx/ogg.conf:1]: This should be `2024`

File: OracleGoldenGate/23/nginx/ogg.conf
Comment (Djelibeybi) [inline OracleGoldenGate/23/nginx/ogg.conf:5]: Something something NGINX something something.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
It occurred to me when I finished the review that all the strings that have spaces between the letters could actually be removed without any loss of context. I figured I'd mention that here instead of going back and editing all those review comments.

Also, the reason why the copyright lines need to be `2024` is because the files themselves are new. The age of the content in those files is irrelevant in this regard.

**Comment by sbalousek** [comment]
Thanks for your time, @Djelibeybi !

**Comment by Djelibeybi** [comment]
> Thanks for your time, @Djelibeybi !

You're welcome. :)

File: OracleGoldenGate/23/Dockerfile
Comment (sbalousek) [reply OracleGoldenGate/23/Dockerfile]: Updated

File: OracleGoldenGate/23/Dockerfile
Comment (Djelibeybi) [inline OracleGoldenGate/23/Dockerfile:4]: I don't know if this bothers you but this line is misaligned with the rest of the file. :)

File: OracleGoldenGate/23/README.md
Comment (Djelibeybi) [inline OracleGoldenGate/23/README.md:179]: The URL should use `https` by default. There is a redirect in place but it would be better not to rely on it.

**Review by Djelibeybi — DISMISSED** [review]
There are two very minor things I spotted but they're far from release blockers. You can decide whether or not you want to address them while the PR is still in draft mode.

File: OracleGoldenGate/23/Dockerfile
Comment (sbalousek) [reply OracleGoldenGate/23/Dockerfile:4]: yes... but then the `BASE_IMAGE` variable name wouldn't align :)

Comment (Djelibeybi) [reply OracleGoldenGate/23/Dockerfile:4]: I hadn't even noticed that `BASE_IMAGE` was on both lines let alone that both instances were aligned. My brain took no notice of what was contained within the eye-twitch-inducing placement of `${...}` 😛 

It's your call either way, so I'll just have to gird my loins by putting on big boy pants if I ever need to review this `Dockerfile` again in future. 😂

**Comment by sbalousek** [comment]
Hi @Djelibeybi - I am having trouble finding the error in the super linter check. Any pointers would be appreciated.

**Comment by Djelibeybi** [comment]
> Hi @Djelibeybi - I am having trouble finding the error in the super linter check. Any pointers would be appreciated.

The linting issues are being found by `SHELL_SHFMT` and can be seen by expanding that section or by clicking here: https://github.com/oracle/docker-images/actions/runs/8926398614/job/24517335048?pr=2780#step:5:233

**Comment by Djelibeybi** [comment]
Though, now that I'm looking at it closely, I'm not sure I agree with the linter on a few things (like its suggested indentation or lack thereof). I'll take a closer look tomorrow or over the weekend and tweak the linter as necessary. If you feel the linter is being overly pedantic, let me know and I'll merge after I do a final manual check.

**Comment by sbalousek** [comment]
I simply do not see the error. The `shfmt` tool is too aggressive with regard to whitespace, marking style differences as errors. I made the suggested corrections, which are still failing for no apparent reason. When I run `shfmt` locally, the return code is 0 for all files using `shfmt -w -i 4`. 

As far as I can tell, this PR failed because `shfmt` expects TAB indentation, which I don't agree with. Other than that, I see no errors.

`shfmt` did, however, find an issue that `ShellCheck` did not.

**Comment by sbalousek** [comment]
These changes - which are critical to the GoldenGate 23ai release - passed lint checks a few weeks ago. If there is anything you can do before next week, I would really appreciate it.

**Comment by sbalousek** [comment]
I thought about this more and realized TAB characters are the current requirement (per `shfmt`), so I reformatted and tested the scripts.

**Comment by sbalousek** [comment]
Thanks a bunch, Avi. Have a fantastic weekend!

**Comment by Djelibeybi** [comment]
> Thanks a bunch, Avi. Have a fantastic weekend!

You're welcome and likewise.

**Pull request merged**
